CMakeLists.txt

```diff
@@ -89,7 +84,72 @@ IF (MAKETEST STREQUAL YES)
     ADD_SUBDIRECTORY(trantor/tests)
 ENDIF ()
 
-INSTALL(TARGETS trantor DESTINATION lib)
+SET(public_net_headers
+    trantor/net/EventLoop.h
+    trantor/net/EventLoopThread.h
+    trantor/net/EventLoopThreadPool.h
+    trantor/net/InetAddress.h
+    trantor/net/TcpClient.h
+    trantor/net/TcpConnection.h
+    trantor/net/TcpServer.h
+    trantor/net/callbacks.h)
+    
+SET(public_utils_headers
+    trantor/utils/AsyncFileLogger.h
+    trantor/utils/ConcurrentTaskQueue.h
+    trantor/utils/Date.h
+    trantor/utils/Funcs.h
+    trantor/utils/LockFreeQueue.h
+    trantor/utils/LogStream.h
+    trantor/utils/Logger.h
+    trantor/utils/MsgBuffer.h
+    trantor/utils/NonCopyable.h
+    trantor/utils/ObjectPool.h
+    trantor/utils/SerialTaskQueue.h
+    trantor/utils/TaskQueue.h
+    trantor/utils/TimingWheel.h)
+
+SOURCE_GROUP( "Public API" FILES ${public_net_headers} ${public_utils_headers})
+
+install(TARGETS trantor
+  # IMPORTANT: Add the trantor library to the "export-set"
+  EXPORT TrantorTargets
+  ARCHIVE DESTINATION "${INSTALL_LIB_DIR}" COMPONENT lib)
+
+INSTALL(FILES ${public_net_headers} DESTINATION ${INSTALL_INCLUDE_DIR}/trantor/net)
+INSTALL(FILES ${public_utils_headers} DESTINATION ${INSTALL_INCLUDE_DIR}/trantor/utils)
+
+
+# Add all targets to the build-tree export set
+export(TARGETS trantor
+  FILE "${PROJECT_BINARY_DIR}/TrantorTargets.cmake")
+
+# Export the package for use from the build-tree
+# (this registers the build-tree with a global CMake-registry)
+export(PACKAGE Trantor)
+
+# Create the TrantorConfig.cmake and TrantorConfigVersion files
+file(RELATIVE_PATH REL_INCLUDE_DIR "${INSTALL_TRANTOR_CMAKE_DIR}"
+   "${INSTALL_INCLUDE_DIR}")
+# ... for the build tree
+set(CONF_INCLUDE_DIRS "${PROJECT_SOURCE_DIR}" "${PROJECT_BINARY_DIR}")
+configure_file(cmake/templates/TrantorConfig.cmake.in
```

Comment:
You don't need this, or the `TrantorConfig.cmake.in` file, CMake can generate it fully.  Above on the `export` line it can autogenerate it.  The basic format of those is:
```
export(TARGETS
        your_target
    NAMESPACE YourPackage::
    FILE "${CMAKE_CURRENT_BINARY_DIR}/YourPackageConfig.cmake"
)
install(EXPORT
        YourPackageConfig
    DESTINATION "${CMAKE_INSTALL_DATADIR}/YourPackage/cmake"
    NAMESPACE YourPackage::
)
```

Reply:
You don't need the namespace if you don't have components, but I find it good form to always use it anyway in case components are added later.

Reply:
@OvermindDL1 , I will try it tomorrow. Thank you very much!